assembl/static2/js/app/components/debate/brightMirror/sideComment/sideCommentBox.jsx

```diff
@@ -0,0 +1,333 @@
+    const { extracts } = this.props;
+    return extracts && extracts.length > 0 ? extracts[extractIndex] : null;
+  };
+
+  submit = (): void => {
+    const { postId, contentLocale, lang, addPostExtract, addExtractComment, setCommentBoxDisplay, refetchPost } = this.props;
+    const { body, selectionText, serializedAnnotatorRange } = this.state;
+    if (!selectionText || !serializedAnnotatorRange) {
+      return;
+    }
+
+    const bodyIsEmpty = editorStateIsEmpty(body);
+    if (bodyIsEmpty) {
+      displayAlert('warning', I18n.t('debate.thread.fillBody'));
+      return;
+    }
+
+    const variables = {
+      contentLocale: contentLocale,
+      postId: postId,
+      body: selectionText,
+      important: false,
+      lang: lang,
+      xpathStart: serializedAnnotatorRange.start,
+      xpathEnd: serializedAnnotatorRange.end,
+      offsetStart: serializedAnnotatorRange.startOffset,
+      offsetEnd: serializedAnnotatorRange.endOffset
+    };
+    displayAlert('success', I18n.t('loading.wait'));
+    addPostExtract({ variables: variables })
+      .then((result) => {
+        const postVars = {
+          extractId: result.data.addPostExtract.extract.id,
+          body: convertContentStateToHTML(body.getCurrentContent()),
+          contentLocale: contentLocale
+        };
+        addExtractComment({ variables: postVars }).then(() => {
+          this.setState({
+            submitting: false
+          });
+          setCommentBoxDisplay();
+          window.getSelection().removeAllRanges();
+          displayAlert('success', I18n.t('harvesting.harvestingValidated'));
+          refetchPost();
+        });
+      })
+      .catch((error) => {
+        displayAlert('danger', `${error}`);
+      });
+  };
+
+  hightlightExtract = (extract: FictionExtractFragment) => {
+    const { clearHighlights } = this.props;
+    clearHighlights();
+    const extractElement = document.getElementById(`${extract.id}`);
+    if (extractElement) extractElement.classList.add('active-extract');
+  };
+
+  updateBody = (newValue: EditorState) => {
+    this.setState({
+      body: newValue
+    });
+  };
+
+  renderFooter = () => {
+    const { submitting } = this.state;
+    const actionId = submitting ? ACTIONS.create : ACTIONS.confirm;
+    if (!actionId) return null;
+    const action = this.actions[actionId];
+    return (
+      <div className="harvesting-box-footer">
+        {action.buttons.map(button => (
+          <Button key={button.id} className={`${button.className} button-dark`} onClick={button.onClick}>
+            {I18n.t(button.title)}
+          </Button>
+        ))}
+      </div>
+    );
+  };
+
+  render() {
+    const { contentLocale, extracts, cancelSubmit, position, toggleCommentsBox } = this.props;
+    const { submitting, extractIndex, body } = this.state;
+    const currentExtract = this.getCurrentExtract(extractIndex);
+    if (!submitting && currentExtract) {
+      this.hightlightExtract(currentExtract);
```

Comment:
I don't think it's a good idea to call hightlightExtract here (you have to call it in the didMount or ...)